### PR TITLE
DSD-1101: Header - Adds the third radio button for searching

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes JIRA ticket [DSD-1101](https://jira.nypl.org/browse/DSD-1101)

## This PR does the following:

- Adds a third radio button, separating search options between the Circulating Catalog, Research Catalog, and the NYPL.org website.
- Removes the mobile search buttons in favor of the radio buttons.

NOTE: The URLs for the actual circulating and research catalogs are still pending. That will be a separate PR.

## How has this been tested?

Locally in Storybook.

## Accessibility concerns or updates

- N/A

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] Review the Vercel preview deployment once it is ready.

### CODE REVIEW

File: src/components/Header/components/HeaderSearchForm.test.tsx [hunk 1]
```diff
@@ -116,7 +139,7 @@ describe("HeaderSearchForm", () => {
       // ALL tests. So we should have two calls to `window.location.assign`.
       // The second call to `window.location.assign` should be...
```
Comment (oliviawongnyc) [inline src/components/Header/components/HeaderSearchForm.test.tsx]: Should this be "third call"?

File: src/components/Header/components/HeaderSearchForm.test.tsx [hunk 2]
```diff
@@ -96,6 +97,28 @@ describe("HeaderSearchForm", () => {
...
+      // Select the "Search the Research Catalog" radio button.
+      userEvent.click(researchRadio);
+      userEvent.click(searchBtn);
+
+      // Fast-forward until all timers have been executed.
+      // The SearchForm calls `gaUtils.trackSearchQuerySend` which
+      // internally has a timer.
+      jest.runAllTimers();
+
+      // The first call to `window.location.assign` should be...
```
Comment (oliviawongnyc) [inline src/components/Header/components/HeaderSearchForm.test.tsx]: Should this be "second call"?

File: src/theme/components/header/headerSearchForm.ts [hunk 3]
```diff
@@ -26,16 +21,17 @@ const HeaderSearchForm = {
...
     },
-    fieldset: {
-      margin: { base: "20px 15px 0", md: "0" },
+    "#radio-group-search-type": {
+      margin: { base: "0 20px 20px", md: "0" },
     },
     textInput: {
       label: {
-        fontSize: { md: "18px" },
+        fontSize: "22px",
```
Comment (bigfishdesign13) [inline src/theme/components/header/headerSearchForm.ts]: In the design on the Final V1 page in Figma, the font size for the label is 28px (`heading.secondary`) for desktop and 22px (`heading.tertiary`) for mobile.

https://www.figma.com/file/FH7n0slL59s8Jp0Lmlabv8/Global-Header%3A-3rd-Radio-Button?node-id=1158%3A1177

Comment (bigfishdesign13) [reply src/theme/components/header/headerSearchForm.ts]: Also, I am not sure if there is a style somewhere setting the font size for the radio button, but the label text on the radio buttons is too large.  It is currently 18px (`heading.callout`), but it should be 14px (`text.caption`).

**Review by bigfishdesign13 — CHANGES_REQUESTED** [review]
Overall, it looks good, but there are a few text style issues that need to be cleaned up.  Also, the tab order in the search form currently goes: `text field` —> `submit button` —> `radio group`.  Can that be changed to: `text field` —> `radio group` —> `submit button`?

**Comment by EdwinGuzman** [comment]
Regarding the tab order, the way the form-component layout is rendered, they are rendered as rows and not columns. I don't think this is a difficult change but we might need to build a pattern for this with our current form components. Can we make this a new ticket? Possibly also to update the form components if they need to be updated?

File: src/theme/components/header/headerSearchForm.ts (hunk 3 above)
Comment (EdwinGuzman) [reply src/theme/components/header/headerSearchForm.ts]: I'll make the updates but just a reminder that we were suppose to build the header to look like the production header and not involve too many "new" updates. With this third radio button feature, more style changes are creeping in. This PR probably won't get merged in for a while but just wanted to point this out.

**Review by oliviawongnyc — APPROVED** [review]
Looks good to me! Just a couple very small questions on some of your code comments.

File: src/theme/components/header/headerSearchForm.ts (hunk 3 above)
Comment (bigfishdesign13) [reply src/theme/components/header/headerSearchForm.ts]: OK, I see.  That does get a bit tricky.  However, if UI changes are being introduced into the production header, don’t those same UI changes need to be reflected here as well?

**Comment by bigfishdesign13** [comment]
> Regarding the tab order, the way the form-component layout is rendered, they are rendered as rows and not columns. I don't think this is a difficult change but we might need to build a pattern for this with our current form components. Can we make this a new ticket? Possibly also to update the form components if they need to be updated?

Yes, go ahead and create a separate ticket for this.  Thanks.

**Comment by EdwinGuzman** [comment]
We got final text content for the radio buttons. I updated the text based on https://www.figma.com/file/FH7n0slL59s8Jp0Lmlabv8/Global-Header%3A-3rd-Radio-Button?node-id=1158%3A1178#270612587

**Comment by EdwinGuzman** [comment]
@bigfishdesign13 @oliviawongnyc this is ready for another round of review.

File: src/theme/components/header/headerSearchForm.ts [hunk 4]
```diff
@@ -26,16 +21,17 @@ const HeaderSearchForm = {
...
     },
-    fieldset: {
-      margin: { base: "20px 15px 0", md: "0" },
+    "#radio-group-search-type": {
+      margin: { base: "0 20px 20px", md: "0" },
     },
     textInput: {
       label: {
-        fontSize: { md: "18px" },
+        fontSize: { base: "heading.tertiary", md: "heading.secondary" },
```
Comment (bigfishdesign13) [inline src/theme/components/header/headerSearchForm.ts]: When viewing at the smallest breakpoint (320px), the input label wraps oddly.  I don't think we are going to get a lot of users with that screen size, but let's change `fontSize` to the following to make an accommodation:
```
fontSize: { base: "heading.callout", sm: "heading.tertiary", md: "heading.secondary" },
```

**Review by bigfishdesign13 — CHANGES_REQUESTED** [review]
Overall, I think this looks good.  Just one change requested.

File: src/theme/components/header/headerSearchForm.ts [hunk 5]
```diff
@@ -26,16 +21,21 @@ const HeaderSearchForm = {
...
-      margin: { base: "20px 15px 0", md: "0" },
+    "#radio-group-search-type": {
+      margin: { base: "0 20px 20px", md: "0" },
     },
     textInput: {
       label: {
-        fontSize: { md: "18px" },
+        fontSize: {
+          base: "heading.callout",
+          sm: "heading.tertiary",
```
Comment (bigfishdesign13) [inline src/theme/components/header/headerSearchForm.ts]: This didn't work out as well as I hoped.  Instead, let's drop the `sm` value altogether and just bounce directly from `callout` to `secondary`.
```
fontSize: { base: "heading.callout", md: "heading.secondary" },
```

**Review by bigfishdesign13 — CHANGES_REQUESTED** [review]

**Review by bigfishdesign13 — APPROVED** [review]
Looks good.  Thank you.

**Pull request merged**